### PR TITLE
[Add] DataLoader with virtual point sampling.

### PR DESCRIPTION
**Pull request by ssh98son**
This branch will bring the following features
- DataLoader
- Pick architecture
- Loss (Cross-entropy)
- Training script

Current PR includes:
- DataLoader: I virtually labeled the target point in the observation which is the closest to the object's center.

### CODE REVIEW

File: Simulation/pybullet_env/learning/dataset/policy_dataset.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        # Select target points only
+        pcd = pcd[pcd[:,3]==True]
+        pcd = pcd[:,:3]
+
+
+        return pcd
+
+
+
+class PolicyDatasetSE2(Dataset):
```
Comment (ajy8456) [inline Simulation/pybullet_env/learning/dataset/policy_dataset.py:81]: Why did you name it SE2 separately? Isn't it compatible with SE3? I think it will be compatible with SE3.

File: Simulation/pybullet_env/learning/dataset/policy_dataset.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+            list_observation_pcd.append(observation_pcd)
+            list_reward.append(reward)
+
+        # Select pick action and corresponding observation
+        list_prev_observation_pcd = deepcopy(list_observation_pcd)
+        list_prev_observation_pcd.insert(0, init_observation_pcd)
+        trajectory_length = len(trajectory_action)
+        while True:
+            # Randomly select the time step.
+            i = np.random.randint(0, trajectory_length)
```
Comment (ajy8456) [inline Simulation/pybullet_env/learning/dataset/policy_dataset.py:195]: If you sample the time step in the `trajectory_length` without any conditions, it seems that you cannot distinguish between when the target action is PICK and when it is PLACE. You should only choose when the target actions are PICK.

**Review by ajy8456 — CHANGES_REQUESTED** [review]

File: Simulation/pybullet_env/learning/model/policy/pointpick.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+        self.actvn = nn.LeakyReLU(0.2)
+        self.pool = maxpool
+
+
+    def forward(self, p):
+        """
+        Args: 
+            p: (B, num_points, dim)
+        """
```
Comment (ajy8456) [inline Simulation/pybullet_env/learning/model/policy/pointpick.py:48]: Please leave comment for returns.

File: Simulation/pybullet_env/learning/dataset/policy_dataset.py (hunk 2 above)
Comment (ssh98son) [reply Simulation/pybullet_env/learning/dataset/policy_dataset.py:195]: The wrapping while loop and if condition will filter the "PICK". Please see lines 190~210. If those are confusing, please make an additional comment.

File: Simulation/pybullet_env/learning/dataset/process_policy_data.py [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+        # Select target points only
+        pcd = pcd[pcd[:,3]==True]
+        pcd = pcd[:,:3]
+
+
+        return pcd
+
+
+
+class PolicyDataset(Dataset):
```
Comment (ajy8456) [inline Simulation/pybullet_env/learning/dataset/process_policy_data.py:81]: I think this name could be overlap with another dataset which contain belief. Isn't it better to change the name to notice that inputs of this dataset is observation.

File: Simulation/pybullet_env/learning/dataset/process_policy_data.py [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+if __name__ == '__main__':
+
+    # Debugging scripts
+    BATCH_SIZE = 4
+    data_path = os.path.join(os.path.expanduser("~"), "vessl", "deprecate", "no_state", "sim_dataset")
+
+    # Dataset
+    dataset = PolicyDatasetSE2(
+        config = PolicyDatasetSE2.Config(),
```
Comment (ajy8456) [inline Simulation/pybullet_env/learning/dataset/process_policy_data.py:250]: You should change the class name. The name of dataset class is not `PolicyDatasetSE2` anymore.

**Review by ajy8456 — CHANGES_REQUESTED** [review]

**Review by ajy8456 — APPROVED** [review]
